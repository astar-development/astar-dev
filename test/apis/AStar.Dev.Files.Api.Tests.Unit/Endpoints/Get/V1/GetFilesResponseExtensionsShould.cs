using AStar.Dev.Infrastructure.FilesDb.Models;
using AStar.Dev.Utilities;

namespace AStar.Dev.Files.Api.Endpoints.Get.V1;

public class GetFilesResponseExtensionsShould
{
    // [Fact]
    // public void MapTheDuplicateDetailsAsExpected()
    // {
    //     var duplicateDetail = new DuplicateDetail
    //                           {
    //                               DirectoryName     = "MockDirectoryName",
    //                               FileName          = "MockFileName",
    //                               FileHandle        = "MockFileHandle",
    //                               FileSize          = 12345,
    //                               FileLastViewed    = new (2025, 12, 23, 0, 1, 2, TimeSpan.Zero),
    //                               HardDeletePending = new (2025, 12, 23, 0, 2, 2, TimeSpan.Zero),
    //                               SoftDeletePending = new (2025, 12, 23, 0, 3, 2, TimeSpan.Zero),
    //                               SoftDeleted       = new (2025, 12, 23, 0, 4, 2, TimeSpan.Zero),
    //                               UpdatedOn         = new (2025, 12, 23, 0, 5, 2, TimeSpan.Zero),
    //                               ImageHeight       = 123,
    //                               ImageWidth        = 456,
    //                               Instances         = 1,
    //                               IsImage           = true
    //                           };
    //
    //     var sut = duplicateDetail.ToGetFilesResponse();
    //
    //     sut.ToJson().ShouldMatchApproved();
    // }

    [Fact]
    public void MapTheFileDetailsAsExpected()
    {
        var fileDetail = new FileDetail
                         {
                             DirectoryName  = "MockDirectoryName",
                             FileName       = "MockFileName",
                             FileHandle     = "MockFileHandle",
                             FileSize       = 12345,
                             FileLastViewed = new (2025, 12, 23, 0, 1, 2, TimeSpan.Zero),
                             DeletionStatus = new()
                                              {
                                                  HardDeletePending = new (2025, 12, 23, 0, 2, 2, TimeSpan.Zero),
                                                  SoftDeletePending = new (2025, 12, 23, 0, 3, 2, TimeSpan.Zero),
                                                  SoftDeleted       = new (2025, 12, 23, 0, 4, 2, TimeSpan.Zero)
                                              },
                             UpdatedOn   = new (2025, 12, 23, 0, 5, 2, TimeSpan.Zero),
                             ImageDetail = new(123, 456),
                             IsImage     = true
                         };

        var sut = fileDetail.ToGetFilesResponse();

        sut.ToJson().ShouldMatchApproved();
    }
}
