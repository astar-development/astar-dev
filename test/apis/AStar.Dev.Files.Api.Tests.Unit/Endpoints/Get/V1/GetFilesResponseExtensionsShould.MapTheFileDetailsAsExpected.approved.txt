{
  "directoryName": "MockDirectoryName",
  "fileName": "MockFileName",
  "fileHandle": "MockFileHandle"
}